SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -236,51 +349,63 @@ void SpatialEventTracer::RemoveComponent(Worker_EntityId EntityId, Worker_Compon
 	EntityComponentSpanIds.Remove({ EntityId, ComponentId });
 }
 
-void SpatialEventTracer::UpdateComponent(Worker_EntityId EntityId, Worker_ComponentId ComponentId, const Trace_SpanId& SpanId)
+void SpatialEventTracer::UpdateComponent(Worker_EntityId EntityId, Worker_ComponentId ComponentId, const FSpatialGDKSpanId& SpanId)
 {
-	if (Trace_SpanId_Equal(SpanId, Trace_SpanId_Null()))
+	if (!SpanId.IsValid())
 	{
 		return;
 	}
 
-	Trace_SpanId& StoredSpanId = EntityComponentSpanIds.FindChecked({ EntityId, ComponentId });
+	FSpatialGDKSpanId& StoredSpanId = EntityComponentSpanIds.FindChecked({ EntityId, ComponentId });
 
-	Trace_SpanId MergeCauses[2] = { SpanId, StoredSpanId };
-	TOptional<Trace_SpanId> NewSpanId = CreateSpan(MergeCauses, 2);
+	FMultiGDKSpanIdAllocator SpanIdAllocator = FMultiGDKSpanIdAllocator(SpanId, StoredSpanId);
```

Comment:
Hmm, if the invalid/null states were merged this could go back to the stack-allocated array because `FSpatialGDKSpanId` would match the previous Trace_SpanId exactly

Reply:
Sorry, I don't follow. If two null states are merged, we it would be favorable to use the stack array because?